WebCryptoAPI/derive_bits_keys/cfrg_curves_bits.js

```diff
@@ -101,11 +108,8 @@ function define_tests() {
 
           // - wrong algorithm
           promise_test(function(test) {
-              publicKey = publicKeys["X25519"];
-              if (algorithmName === "X25519") {
-                  publicKey = publicKeys["X448"];
-              }
-              return subtle.deriveBits({name: algorithmName, public: publicKey}, privateKeys[algorithmName], 8 * sizes[algorithmName])
+              publicKey = publicKeys[algorithmName];
+              return subtle.deriveBits({name: "ECDH", public: publicKeys[algorithmName]}, privateKeys[algorithmName], 8 * sizes[algorithmName])
```

Comment:
Like this, the test doesn't really test much because the `ECDH` algorithm also requires a `namedCurve`, for example. So I would keep it as it was before and use an X448 public key with X25519 and vice versa.

Reply:
The problem ins that since X488 is not implemented, it gives us quite some noise.

Reply:
OK, yeah.

As written, I think this is more like a test of ECDH than of X25519 (though it might be an interesting test to have as well), but it could work if you pass an ECDH key to `public` parameter, perhaps?

Reply:
The test is about passing a wrong 'algorithm' parameter to the deriveBits operation, to test this step:

> 4. If the [name](https://www.w3.org/TR/WebCryptoAPI/#dom-keyalgorithm-name) attribute of the [[[algorithm]]](https://www.w3.org/TR/WebCryptoAPI/#dfn-CryptoKey-slot-algorithm) internal slot of publicKey is not equal to the [name](https://www.w3.org/TR/WebCryptoAPI/#dom-keyalgorithm-name) property of the [[[algorithm]]](https://www.w3.org/TR/WebCryptoAPI/#dfn-CryptoKey-slot-algorithm) internal slot of key, then [throw](https://webidl.spec.whatwg.org/#dfn-throw) an [InvalidAccessError](https://webidl.spec.whatwg.org/#invalidaccesserror).

So a valid algorithm should be passed there, in order to ensure we reach the X25519 check specified in '4.'.  I used ECDH because it was the most similar one to X25519, available today. It'd be better to use X448, but since it's not implemented in any browser, the test is not useful as it is now. 

Maybe there is a better alternative ?

Reply:
But if you pass "ECDH" as the algorithm, we never end up in that step of the X25519 algorithm, you end up in the ECDH algorithm (which has the same step, but that's not what we're trying to test here).

Hence "X25519" should be passed as the algorithm, with (e.g.) an ECDH public key in the `public` property.

Reply:
True, I've just realized about that mistake. Fixed it now.